deflate_slow.c

```diff
@@ -119,7 +119,7 @@ ZLIB_INTERNAL block_state deflate_slow(deflate_state *s, int flush) {
     }
     Assert(flush != Z_NO_FLUSH, "no flush?");
     if (s->match_available) {
-        bflush = zng_tr_tally_lit(s, s->window[s->strstart-1]);
+        (void) zng_tr_tally_lit(s, s->window[s->strstart-1]);
```

Comment:
Why cast return value to void? It doesn't seem necessary. Am I missing something?

Reply:
Casting to void is the traditional way of telling static analyzers (and DoD code reviewers) that you intend to ignore the return value.

I'm having trouble believing this isn't well known, perhaps I'm showing my age :-)

Reply:
I know about that with regard to variables but I’ve not seen it used with regard to return values from functions which are not stored. I think it would only need to be (void)bflush if you intended to assign to bflush but since you are not then there is no need to cast to void??

Reply:
*shrug* That's the way it works.  Unless you annotate it somehow, the static analyzer / DoD code reviewer has no way of knowing you intended to ignore that return value.

See also https://lmgtfy.com/?q=casting+function+return+value+to+void

Reply:
I see ok